lib/logstash/outputs/s3.rb

```diff
@@ -106,6 +109,13 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   # S3 bucket
   config :bucket, :validate => :string, :required => true
 
+  # Specify a custom endpoint for use with non-AWS S3 implementations, e.g.,
+  # Ceph.  Provide a URL in the format http://127.0.0.1:8080/
+  config :endpoint, :validate => :string
+
+  # When false, specify the bucket in the subdomain.  When true, specify the bucket in the path.
+  config :force_path_style, :validate => :bool, :default => false
```

Comment:
"force path style" feels like the wrong name for this setting. 

It feels weird to have users specify the bucket in the path. Here are some citations of existing examples:

* hadoop uses the "host" part of a URI for the bucket, such as `s3a://bucket-name/...`
* s3cmd lets you specify how to take a bucket name and turn it into a hostname (host_bucket)

The name of the setting "force path style" doesn't indicate clearly enough what it does. We need a more informative name for this setting. Alternately, we can step back and look at the problem space from a different perspective and maybe come up with a solution that doesn't need this kind of flag.

`:bool` is not a correct validation. Did you mean `:boolean` ?